src/v/kafka/types.h

```diff
@@ -91,6 +92,33 @@ using leader_epoch = named_type<int32_t, struct leader_epoch_tag>;
  */
 static constexpr leader_epoch invalid_leader_epoch(-1);
 
+struct uuid {
+    static constexpr auto length = 16;
+
+    template<typename Consumer>
+    void set_uuid(Consumer& c) {
+        c.consume(length, [this](const char* src, size_t max) {
+            std::copy_n(src, max, _uuid.data());
+            return ss::stop_iteration::no;
+        });
+    }
+
+    /// implicit conversions to string_view
+    std::string_view view() const {
+        return std::string_view(
+          reinterpret_cast<const char*>(_uuid.data()), _uuid.size());
+    }
```

Comment:
It feels strange to have `view()` and `to_string` return different values.
```suggestion
    bytes_view() view() const {
        return {reinterpret_cast<const char*>(_uuid.data()), _uuid.size());};
    }
```

Reply:
Yeah i agree, however thats how the official one [works](https://home.apache.org/~vvcephei/kafka-2.8.0-rc2/javadoc/org/apache/kafka/common/Uuid.html). We don't need to copy that behavior of course

Reply:
Oh, I see. I think I misunderstood what `base64_to_string` does. It doesn't take base64 encoded bytes to a string, it takes bytes to a base64 encoded string. That's fine.

I think use the `bytes_view` for `view()` and `bytes_to_base64(view())`, which at least makes sense when I read it.